tedana/workflows/tedana.py

```diff
@@ -705,6 +706,9 @@ def tedana_workflow(
             # If we're going to restart, temporarily allow force overwrite
             if keep_restarting:
                 io_generator.overwrite = True
+                # Create a re-initialized selector object if rerunning
+                selector = ComponentSelector(tree)
```

Comment:
This shouldn't be necessary. If it is, then we should open an issue to fix it in a future PR.

Reply:
This was necessary because `selector.select` looks for the last node in the decision tree that was run and then continues to run the rest of the nodes. This is how `ica_reclassify` works.

This is already happening in `main` but it is within `tedica.automatic_selection`

https://github.com/ME-ICA/tedana/blob/a21d65ec5b1682a52cb8fe0eee1689d5eb3154a3/tedana/selection/tedica.py#L69-L70

By removing the component table from the `selector` initialization and now initializing in `tedana.py` the fact that this was happening is more up-front.

If the goal is to log all failed attempts, we would probably want to save both the ica mixing matrices and the outputs. This is a bit tricky because we currently save most files at the end and not as the data in those files are generated. We could also add some way to tack on a full tree to the end of the executed tree, but that might just add confusion. Either way, this does feel like a separate PR.

Reply:
Ah, thank you for explaining. This is different from scikit-learn classes, which have `fit`, `fit_transform`, and `transform` methods that distinguish between fitting the estimator to training data and transforming new data using the parameters from the fitting data. I know we don't need to have one-to-one correspondence, but I guess I didn't intuitively grasp it.

Reply:
I'd say this is the equivalent of running `fit_transform` a second time on the same dataset using the slightly different parameters (different seed in our cause). We also have a second use case that's the equivalent of running a second transform on a dataset that's already been transformed. I'm not sure that's something that is or should be time with the methods in `scikit-learn`